src/Test.UnitTests.Sarif.Driver/PathExtensionsTests.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+using FluentAssertions;
+
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.CodeAnalysis.Sarif.Driver;
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Test.UnitTests.Sarif.Driver
+{
+    public class PathExtensionsTests
+    {
+        private static readonly int randomSeed = (new Random()).Next();
+        private static readonly Random random = new Random(randomSeed);
+
+        private readonly ITestOutputHelper _outputHelper;
+
+        public PathExtensionsTests(ITestOutputHelper outputHelper)
+        {
+            this._outputHelper = outputHelper;
+            outputHelper.WriteLine($"TestName: {nameof(PathExtensionsTests)} has random seed {randomSeed}");
+        }
+
+        [Theory]
```

Comment:
>Theory [](http://example.com/codeflow?start=9&length=6)

Please stop introducing [Theory] usage into the code base, this functionality uses reflection and historically has caused performance issues.

All this is really buying is is the need not to inject a foreach in the method that calls into your helper.

Reply:
changed to Fact pattern